src/main/java/org/openstreetmap/atlas/checks/validation/tag/StreetNameCheck.java

```diff
@@ -0,0 +1,339 @@
+ * Flags different values (ie "ss", "\u00df", or deprecated tags) for each country specified. Config
+ * file contains all the tags required to tag or not to tag specified for each country. The check
+ * looks at the object's ISO code and then pulls all the needed tags for that country from
+ * configuration.json .
+ *
+ * @author v-naydinyan
+ */
+public class StreetNameCheck extends BaseCheck<Long>
+{
+
+    /**
+     * The object of class CountryInfo contains all the tags and values needed to be flagged for the
+     * object.
+     */
+    class CountryInfo
+    {
+        private List<String> tagContains;
+        private List<String> tagNotContains;
+        private List<String> deprecatedTags;
+        private List<String> values;
+
+        CountryInfo(final List<String> contains, final List<String> notContains,
+                final List<String> deprecated, final List<String> tags)
+        {
+            this.tagContains = contains;
+            this.tagNotContains = notContains;
+            this.deprecatedTags = deprecated;
+            this.values = tags;
+        }
+
+        public List<String> getDeprecatedTags()
+        {
+            return this.deprecatedTags;
+        }
+
+        public List<String> getTagContains()
+        {
+            return this.tagContains;
+        }
+
+        public List<String> getTagNotContains()
+        {
+            return this.tagNotContains;
+        }
+
+        public List<String> getVals()
+        {
+            return this.values;
+        }
+    }
+
+    private static final List<String> ALL_COUNTRIES_ISO_DEFAULT = List.of("AUT", "CHE", "DEU",
+            "LIE");
+    private static final List<List<String>> ALL_VALUES_TO_FLAG_DEFAULT = List.of(List.of("strasse"),
+            List.of("straße"), List.of("strasse"), List.of("straße"));
+    private static final List<List<String>> ALL_VALUES_NOT_FLAG_DEFAULT = List
+            .of(List.of("strasser"), List.of(), List.of("strasser"), List.of());
+    private static final List<List<String>> ALL_DEPRECATED_VALUES_TO_FLAG_DEFAULT = List
+            .of(List.of(), List.of(), List.of("associatedstreet"), List.of());
+    private static final List<List<String>> ALL_ITEMS_TO_FLAG_DEFAULT = List.of(List.of("ss"),
+            List.of("ß"), List.of("ss"), List.of("ß"));
+
+    private static final String CONTAINS_VALUE_INSTRUCTION = "The object contains flagged tags: {0}";
+    private static final String CONTAINS_DEPRECATED_VALUE_INSTRUCTION = "The type tag {0} is deprecated.";
+    private static final List<String> FALLBACK_INSTRUCTIONS = Arrays
+            .asList(CONTAINS_VALUE_INSTRUCTION, CONTAINS_DEPRECATED_VALUE_INSTRUCTION);
+
+    private final List<String> allCountriesIsoConfig;
+    private final List<List<String>> allValuesToFlagConfig;
+    private final List<List<String>> allValuesNotFlagConfig;
+    private final List<List<String>> allDeprecatedValuesToFlagConfig;
+    private final List<List<String>> allItemsToFlagConfig;
```

Comment:
This never seems to be functionally used, even though these values are supposed to be the main reason for flagging objects.